app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala

```diff
@@ -85,4 +88,16 @@ class UpdateNextYearsIncomeService {
       updatedValue
     }
   }
+
+  def submit(employmentId: Int, nino: Nino)(implicit hc: HeaderCarrier): Future[TaiResponse] = {
+    get(employmentId, nino) flatMap {
+      case UpdateNextYearsIncomeCacheModel(employmentName, _, _, Some(newValue)) => {
+        taxAccountService.updateEstimatedIncome(nino, newValue, TaxYear().next, employmentId)
+      }
+      case _ => {
+        // TODO: Should this be something more meaningful?
```

Comment:
This this the sort of place where you want to redirect back to the edit page? There is not real reason why they should ever be here without a valid cache

Reply:
This is the same thing I was thinking with https://github.com/hmrc/tai-frontend/pull/241/files#diff-cc93f81a2f6a026fd7936a9da58b5828R235